src/installer.redmod.ts

```diff
@@ -313,50 +488,76 @@ export const testForREDmod: V2077TestFunc = (
+  (detectToplevelREDmodLayout(fileTree)
+    ? some(toplevelLayoutDir)
+    : none);
+
+
 export const installREDmod: V2077InstallFunc = async (
   api: VortexApi,
   fileTree: FileTree,
   modInfo: ModInfo,
-  _features: Features,
+  features: Features,
 ): Promise<VortexInstallResult> => {
-  //
-  const allPossibleRedmodLayouts = [
-    canonREDmodLayout,
-    namedREDmodLayout,
-    toplevelREDmodLayout,
-  ];
-
-  const selectedInstructions = await useFirstMatchingLayoutForInstructionsAsync(
-    api,
-    modInfo.name,
-    fileTree,
-    modInfo.installingDir,
-    allPossibleRedmodLayouts,
+  const singleModPipeline =
+    (relativeModDir: string): TaskEither<Error, readonly VortexInstruction[]> =>
+      pipe(
+        tryReadInfoJson(modInfo.installingDir, relativeModDir),
+        chainEitherKW((redmodInfo) => pipe(
+          collectPathDetesForInstructions(relativeModDir, redmodInfo, fileTree),
+          chainE((modInfoAndPathDetes) => pipe(
+            [
+              initJsonLayoutAndValidation,
+              archiveLayoutAndValidation,
+              customSoundLayoutAndValidation,
+              scriptLayoutAndValidation,
+              tweakLayoutAndValidation,
+              extraFilesLayoutAndValidation,
+            ],
+            traverseArrayE((layout) => layout(api, modInfoAndPathDetes)),
+          )),
+          mapE(flatten),
+        )),
+      );
+
+  const allModsForLayoutPipeline = pipe(
+    [
+      canonLayoutMatches,
+      namedLayoutMatches,
+      toplevelLayoutMatches,
+    ],
+    findFirstMap((allModDirsForLayoutIfMatch) => allModDirsForLayoutIfMatch(fileTree)),
+    fromOptionTE(knownError(`No REDmod layout found! This shouldn't happen, we already tested we should handle this!`)),
+    chainEitherK((allModDirsForLayout) => allModDirsForLayout(fileTree)),
+    chain(flow(
+      traverseArrayTE(singleModPipeline),
+      mapTE(flatten),
+    )),
   );
 
-  if (
-    selectedInstructions === NoInstructions.NoMatch ||
-    selectedInstructions === InvalidLayout.Conflict
-  ) {
-    const errorMessage = `Didn't Find Expected REDmod Installation!`;
-
-    api.log(
-      `error`,
-      `${InstallerType.REDmod}: ${errorMessage}`,
-      sourcePaths(fileTree),
-    );
-
-    showWarningForUnrecoverableStructureError(
-      api,
-      InstallerType.REDmod,
-      errorMessage,
-      sourcePaths(fileTree),
-    );
-
-    return Promise.reject(new Error(errorMessage));
-  }
+  // At this point we have to break out to interop with the rest..
+  const allInstructionsForEverySubmodInside = await allModsForLayoutPipeline();
 
-  return Promise.resolve({ instructions: selectedInstructions.instructions });
+  return isLeft(allInstructionsForEverySubmodInside)
+    ? failAfterWarningUserAndLogging(api, fileTree, modInfo, features, allInstructionsForEverySubmodInside.left)
+    : returnInstructionsAndLogEtc(api, fileTree, modInfo, features, allInstructionsForEverySubmodInside.right);
 };
```

Comment:
```suggestion
};

```
Maybe it is the ruby dev in me, but I like to have the empty last line for some reason. No need to enforce it though, just commenting here